sw/host/opentitanlib/src/transport/ti50emulator/uart.rs

```diff
@@ -56,15 +50,14 @@ impl Ti50Uart {
     }
 
     pub fn get_socket(&self) -> Result<RefMut<UnixStream>> {
-        self.reconnect()?;
         return Ok(RefMut::map(self.socket.borrow_mut(), |socket| {
             socket.as_mut().unwrap()
         }));
     }
 }
 
 /// A trait which represents a UART.
-impl Uart for Ti50Uart {
+impl Uart for Rc<Ti50Uart> {
```

Comment:
In this CL, I wanted to have `Rc<dyn Uart>` and `Rc<Ti50Uart>` simutaneously referring to the same object.  (The former for returning from the `Transport` trait, and the latter for allowing logic internal to the transport relating to starting a new sub-process to iterate over Uart ports, and call an internal method on each.

The way I found to allow the two kinds of Rc's was to not have the `Uart` trait implemented directly by `Ti50Uart`, but by the wrapped `Rc<Ti50Uart>`.  Apparently, Rust implicitly deferences all the mentions of `self` in the implementing code.  I could want to use a similar idiom in other transport backends, which may want to have internal methods, accessible from elsewhere within the transport implementation.

Reply:
@cfrantz 
Can you look at that Rust language construction ?

Reply:
Chris has explained to me that contrary to my previous belief and statements, you can in fact have `Rc<dyn Uart>` and `Rc<Ti50Uart>` simultaneously referring to the same struct.  I have updated the code accordingly, avoiding implementing a trait on `Rc<Ti50Uart>`.